packages/playwright/src/common/fixtures.ts

```diff
@@ -193,12 +200,29 @@ export class FixturePool {
 
     const hash = crypto.createHash('sha1');
     const names = Array.from(this._registrations.keys()).sort();
+
+    // First iterate over non-boxed fixtures to provide clear error messages.
     for (const name of names) {
       const registration = this._registrations.get(name)!;
-      visit(registration);
+      if (registration.box)
+        continue;
+      visit(registration, true);
       if (registration.scope === 'worker')
         hash.update(registration.id + ';');
     }
+
+    // If no errors found, iterate over boxed fixtures
+    if (!hasDependencyErrors) {
+      for (const name of names) {
+        const registration = this._registrations.get(name)!;
+        if (!registration.box)
+          continue;
+        visit(registration, false);
+        if (registration.scope === 'worker')
+          hash.update(registration.id + ';');
```

Comment:
I think it is important to call `hash.update()` for all worker fixtures, even when `hasDependencyErrors`.